### ISSUE TITLE
In kernel::terminate() the response object is filled ok but blank screen displayed

### ISSUE THREAD

**Issue by fredpeaks** [symfony/symfony#14328]
Hi,

On some pages of my website I have a blank screen displayed.
The problem only occurs on my vagrant vm (the site has been tested with the same data on a some servers and this blank problem never happened).
And it's only present in dev environment with debug. If I change app_dev.php to put ```$kernel = new AppKernel('dev', false);```, the page is displayed correctly without error.
When I do a step-by-step debug with PhpStorm, I can see that the ```$response``` object in ```$kernel->terminate($request, $response);``` (last line of app_dev.php) is filled with the expected and correct html content.
So the problem happens inside the terminate call but I'm not able to find what it is.
I think this is a server-configuration related error but it cannot be displayed correctly because of the Symfony error handler.
In the debugger, I can see that ```Symfony/Component/Debug/ErrorHandler::handleFatalError()``` is called after ```$kernel->terminate($request, $response);``` but ```$error``` is null, even after ```$error = error_get_last();```
And there is no error in the php error_log file.

Conditions:
- symfony 2.6.6
- vagrant / puphpet with a Debian 7 distribution
- php 5.6.7-1~dotdeb.2 
- PHP SAPI: fpm-fcgi

Thank you for your help.

Ps: I'd like to get rid of the Symfony2 error handler in order to be able to see "original" php errors but I can't find how. It's only for debugging of course. If someone has a solution, it would be great.

**Comment by fredpeaks** [comment]
After some more research with debugger, I think the problem may come from fastcgi or php-fpm because the 500 error response is thrown from the ```Symfony/Component/HttpFoundation/Response::sendContent()``` call.
This method only does a ```echo $this->content;```. I've checked that content is correct before the ```echo``` call but the browser receives instantly an empty 500 response. And of course if I just do a ```echo``` of the same content in a simple php script it works fine.
I also tried to remove the ```echo``` call or replace it with ```echo '';``` and this time the browser receives the empty 500 response after the ```fastcgi_finish_request();``` in ```Symfony/Component/HttpFoundation/Response::send()```.

I don't know were to look, there isn't anything in the logs (apache, php-fpm, php, symfony, ...) and I can't see any option for php-fpm, fast_cgi, apache or php that could help avoid this problem.

**Comment by nicolas-grekas** [comment]
You should look in your php logs, and/or enable them. The Sf error handler does not hide them at all.

**Comment by fredpeaks** [comment]
@nicolas-grekas I have re-checked that php logs are enabled and they are. I've tested with an "undefined function" error and it appears in php logs. But nothing when I load my blank page. And yeah, I finally understood that the error wasn't hidden by sf error handler as I tried to explain in my previous post. But I cannot find were the problem comes from. Thanks

**Comment by nicolas-grekas** [comment]
If you have xdebug, can you try with scream mode on?

**Comment by fredpeaks** [comment]
@nicolas-grekas I've set xdebug.scream to on but still no error message. (thx for the tip, I didn't knew the scream option)

**Comment by fredpeaks** [comment]
I've tried to remove some fields on a form that is rendered on one of the pages that display a blank screen.
The page has been displayed correctly (without the fields of course).
But the problem doesn't come from these particular fields because the page display correctly when I put them back and remove some other fields.
I think the pages that are concerned by the problem are the most complex in term of database queries, form fields, objects involved, ...
I've updated the xdebug.max_nesting_level to 10000 in case it would be it but no improvement.

**Comment by nicolas-grekas** [comment]
Can you set it to -1 instead of 512M?

**Comment by fredpeaks** [comment]
@nicolas-grekas I've tried -1 instead of 512M but still the same problem. I know the problem doesn't come from memory because this site has been tested on several servers (windows or linux) and never has memory problem. It has no error of any type at all on other servers, it's working fine and fast...

**Comment by nicolas-grekas** [comment]
The ultimate tool are xdebug traces: http://xdebug.org/docs/execution_trace
Look at the output file and where it stops...

**Comment by fredpeaks** [comment]
@nicolas-grekas I'm back after some tests with xdebug traces and it appears that there is no error in the trace, the script doesn't stop until "Symfony\Component\HttpKernel\DataCollector\DumpDataCollector->__destruct()" which is the normal way a Symfony script ends. By the way, the trace file weight 76.5Mo in the most compact form, it's a bit scary !  Here is a compressed version (3Mo), just in case you want to have a look : http://wikisend.com/download/424632/trace.2043925204.xt.tar.gz).
So the problem must be related to something else than php/symfony. I think php-fpm or apache...

**Comment by fredpeaks** [comment]
Ok, I finally found the origin of the problem.
There was an apache error log I didn't see before (*34y6vt7aepki_error_ssl.log*) were I found these types of messages : ```proxy_fcgi:error``` with the text ```Premature end of script headers: app_dev.php, ...```
This leads me to a stackoverflow page where the solution was : http://stackoverflow.com/questions/28602857/after-login-get-blank-page-only-in-chrome

The problem is that the limit size of the headers in my server is too small and the chromephp logging leads to a silent error because of that...
I've disabled chromephp and firefoxphp logging and it seems to work now. I will search how to increase this headers limit in apache.

Thanks for the help @nicolas-grekas
